### PR TITLE
implemented task

### PR DESCRIPTION
**Pull request by ihorvasylenko23**

### CODE REVIEW

File: src/test/java/core/basesyntax/ActivityStrategyImplTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import core.basesyntax.service.strategy.BalanceActivityHandler;
+import core.basesyntax.service.strategy.PurchaseActivityHandler;
+import core.basesyntax.service.strategy.ReturnActivityHandler;
+import core.basesyntax.service.strategy.SupplyActivityHandler;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ActivityStrategyImplTest {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/ActivityStrategyImplTest.java:17]: maybe you should test behavior when argument is null

File: src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import core.basesyntax.service.strategy.ReturnActivityHandler;
+import core.basesyntax.service.strategy.SupplyActivityHandler;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ActivityWorkerServiceImplTest {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java:28]: Maybe you should add more tests, like empty list or null

File: src/test/java/core/basesyntax/FruitTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.model.Fruit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FruitTest {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/FruitTest.java:10]: U don`t need tests for models, only for class which represent some logic, like service

File: src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.service.strategy.PurchaseActivityHandler;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PurchaseActivityHandlerTest {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java:9]: where test with valid result?

File: src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    private static final int QUANTITY_AFTER = 15;
+    private PurchaseActivityHandler purchaseActivityHandler;
+
+    @BeforeEach
+    public void setUp() {
+        purchaseActivityHandler = new PurchaseActivityHandler();
+    }
+
+    @Test
+    public void testQuantityModify_NotEnoughFruitToPurchase_notOk() {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java:20]: if u use "_notOk" only it 1 test, remove it

File: src/test/java/core/basesyntax/ReportCreatorServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.impl.ReportCreatorServiceImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ReportCreatorServiceImplTest {
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/ReportCreatorServiceImplTest.java:14]: maybe u need more tests, for example with empty fruitData

**Review by keda-vlad — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java [hunk 7]
```diff
@@ -61,6 +63,19 @@ public void testModifyQuantity_Success() {
         assertEquals(FIRST_EXP_QUANTITY, fruitDao.getByName(FIRST_FRUIT_NAME));
     }
 
+    @Test
+    public void testModifyQuantity_EmptyList() {
+        assertThrows(RuntimeException.class,
+                () -> activityWorkerService.modifyQuantity(Collections.emptyList()));
+
```
Comment (keda-vlad) [inline src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java:70]: Redundant empty line

**Review by keda-vlad — APPROVED** [review]
Good job, only remove redundant empty line

File: src/test/java/core/basesyntax/ActivityStrategyImplTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        try {
+            activityStrategy.getQuantityModifier(null);
+            fail("Expected IllegalArgumentException, but got no exception");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Activity type cannot be null", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testGetQuantityModifier_Success() {
```
Comment (Vasul-Pankiv) [inline src/test/java/core/basesyntax/ActivityStrategyImplTest.java:38]: incorrect naming for test (fix it for all your test)
see [common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop-tests#remember-about-naming-of-test-methods)

File: src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java (hunk 7 above)
Comment (Vasul-Pankiv) [reply src/test/java/core/basesyntax/ActivityWorkerServiceImplTest.java:70]: not resolved

File: src/test/java/core/basesyntax/CsvReaderServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Test
+    public void testReadFromFile_Success() throws IOException {
+        String filePath = FILE_PATH;
+        createTestFile(filePath, TEST_DATA);
+        List<String> result = csvReaderService.readFromFile(filePath);
+        assertEquals(EXPECTED_SIZE, result.size());
+        assertEquals(FIRST_EXPECTED_DATA, result.get(INDEX_OF_FIRST_EXPECTED_DATA));
+        assertEquals(SECOND_EXPECTED_DATA, result.get(INDEX_OF_SECOND_EXPECTED_DATA));
```
Comment (Vasul-Pankiv) [inline src/test/java/core/basesyntax/CsvReaderServiceImplTest.java:41]: you could check only one line or use list (fix all test)

File: src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.service.strategy.PurchaseActivityHandler;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PurchaseActivityHandlerTest {
```
Comment (Vasul-Pankiv) [inline src/test/java/core/basesyntax/PurchaseActivityHandlerTest.java:10]: Where is tests for other activities?

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — APPROVED** [review]
